kustomize/data_source_kustomization_build.go

```diff
@@ -42,16 +54,39 @@ func dataSourceKustomization() *schema.Resource {
 	}
 }
 
+type kustomizeOptions struct {
+	loadRestrictor string
+}
+
+func getKustomizeOptions(d *schema.ResourceData) (k kustomizeOptions) {
+	// initialize kustomizeOptions with defaults
+	k = kustomizeOptions{
+		loadRestrictor: "",
+	}
+
+	kOpts := d.Get("kustomize_options").(map[string]interface{})
+
+	if kOpts["load_restrictor"] != nil {
+		if kOpts["load_restrictor"].(string) == "none" {
+			k.loadRestrictor = "none"
+		}
+	}
+
+	return k
+}
+
 func kustomizationBuild(d *schema.ResourceData, m interface{}) error {
 	path := d.Get("path").(string)
 
+	kOpts := getKustomizeOptions(d)
```

Comment:
Wouldn't it be better to pass the entire kOpts into runKustomizeBuild and handle the defaults and overwrites in there just once, instead of handling it once for kustomization_build and once for kustomization_overlay?

Reply:
Hi. I'm not sure I follow this here. Do you mean to remove the code for setting the default and checking the load_restictor from `getKustomizeOptions` down into `runKustomizeBuild`?

Reply:
Originally, you changed `runKustomizeBuild` to accept an additiona parameter `load_restrictor string` and passed that string in right from the schema. But now, we're wrapping the `load_restrictor` in a hash map. So you added `getKustomizeOptions` to get the string form the hash map and be able to pass it into the `runKustomizeBuild` method. But inside that method is where it current gets the default config, and if `load_restrictor` was not none, it overwrites that value.

I was wondering, if it wasn't cleaner to pass the hash map into `runKustomizeBuild`, the same way it comes from Terraform, and then handle conversion, defaults and overwriting all just in one place. You wouldn't have to call `getKustomizeOptions` from two places. Plus `getKustomizeOptions` could also handle the defaults and overwriting, that's now implemented in `runKustomizeBuild`.

Basically I'm asking if we should not replace:
```
opts := krusty.MakeDefaultOptions()

// If the load_restrictor option is set to none then the krusty.Options
// should be updated to reflect this.
if load_restrictor == "none" {
    opts.LoadRestrictions = types.LoadRestrictionsNone
}
```

with

```
opts := getKustomizeOptions(kOpts)
```

and change `runKustomizeBuild` to accept `kOpts *schema.ResourceData`

```
runKustomizeBuild(fSys filesys.FileSystem, path string, kOpts *schema.ResourceData)
```

Reply:
But I'm fine merging without that change too. Since all these methods are private, I can probably change this at any time in the future.

Reply:
ok. Passing the schema thru was what i thought you meant originally but then had doubts. That totally makes sense so I will update.

Reply:
OK. Have updated this and its much cleaner now. Having the new `kustomizeOptions` type didnt make any sense also so have removed that.